src/pages/playlists/[slug].tsx

```diff
@@ -8,8 +8,10 @@ export default Playlist
 
 export const getServerSideProps: GetServerSideProps = async ({res, params}) => {
   const slug = params && (params.slug as string)
-  res.setHeader('Location', `/courses/${slug}`)
-  res.statusCode = 302
-  res.end()
-  return {props: {}}
+  return {
+    redirect: {
+      destination: `/courses/${slug}`,
+      permanent: true,
+    },
```

Comment:
this was the issue. the header approach didn't work, sing the redirect syntax does.